### PR TITLE
#164798256 Enable users manipulate articles

### PR DESCRIPTION
**Pull request by patrickf949**
### What does this PR do?

Enables users to create and read articles

#### Description of Task to be completed?

- Add routes for article creation and viewing
- Create pages that users will interact with to create or read stories

#### How should this be manually tested?

Setup the application as stated [here](https://github.com/andela/ah-frontend-prime/pull/1)

Create and verify  a user - [here](https://github.com/andela/ah-frontend-prime/pull/1)

Create articles with the following routes

- `localhost:8000/create-article` --Simulated [here](ah-frontend-prime-pr-15.herokuapp.com/create-article)

View articles on the landing page and click on them to get the full article --Simulated [here](ah-frontend-prime-pr-15.herokuapp.com)

#### What are the relevant pivotal tracker stories?

[164798256](https://www.pivotaltracker.com/story/show/164798256)

#### Screenshots

Create an article
Screenshot 2019-05-20 at 09 42 56

View all articles
Screenshot 2019-05-20 at 03 19 46

### CODE REVIEW

File: src/components/articles/userArticles.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                  <p>
+                    <b>{article.description.substring(0, 60)} ...</b>
+                  </p>
+                  <p>{article.body.substring(0, 60)} ...</p>
+                  <span>
+                    <b>Author: {article.author.username}</b>
+                  </span>
+                  <br />
+                  <span className="card-date">
+                    Created at{" "}
```
Comment (anyatibrian) [inline src/components/articles/userArticles.js:34]: @patrickf949 remove this

File: src/components/articles/createArticlePage.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  onChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+  onSubmit(e) {
+    e.preventDefault();
+    const messageObject = {
```
Comment (e-ian) [inline src/components/articles/createArticlePage.js:57]: @patrickf949 you could do well with destructuring this.

File: src/components/articles/userArticles.js (hunk 1 above)
Comment (patrickf949) [reply src/components/articles/userArticles.js:34]: thanks @anyatibrian, this is necessary to add a space between `Created at` and the time collected from the backend

File: src/actions/LoginAction.js [hunk 3]
```diff
@@ -23,7 +23,10 @@ export const userLoginRequest = userData => async dispatch => {
 
     sessionStorage.setItem("token", response.data.user.token);
     sessionStorage.setItem("username", response.data.user.username);
+    sessionStorage.setItem("email", response.data.user.email);
```
Comment (e-ian) [inline src/actions/LoginAction.js:26]: @patrickf949 why are we storing the email in session here.

File: src/components/articles/createArticlePage.js (hunk 2 above)
Comment (e-ian) [reply src/components/articles/createArticlePage.js:57]: good work with destructuring

File: src/actions/LoginAction.js (hunk 3 above)
Comment (patrickf949) [reply src/actions/LoginAction.js:26]: @e-ian, the email is the most unique way of identifying a user, and that's what we are basing on in the backend to filter user articles

File: src/components/articles/createArticlePage.js (hunk 2 above)
Comment (patrickf949) [reply src/components/articles/createArticlePage.js:57]: Thanks @e-ian

File: src/actions/LoginAction.js (hunk 3 above)
Comment (e-ian) [reply src/actions/LoginAction.js:26]: Okay but the `username` is equally unique in our backend.

**Review by anyatibrian — APPROVED** [review]
GREAT WORK @patrickf949

**Review by e-ian — APPROVED** [review]
LGTM

**Review by Genza999 — APPROVED** [review]
Great work @patrickf949 .

**Review by MuhanguziDavid — CHANGES_REQUESTED** [review]
Great implementation of the articles creation and retrieval. However, the retrieval of articles for a specific user is not working if one is logged in using social auth. This could be because the email is not saved in session storage on social login success. You may need to add it in the social auth action.

**Comment by patrickf949** [comment]
> Great implementation of the articles creation and retrieval. However, the retrieval of articles for a specific user is not working if one is logged in using social auth. This could be because the email is not saved in session storage on social login success. You may need to add it in the social auth action.

Thanks @MuhanguziDavid all done

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Pull request merged**